### ISSUE TITLE
Manager button doesn't appear

### ISSUE THREAD

**Issue by KevenLou321** [Comfy-Org/ComfyUI-Manager#55]
I put the .bat in the right location, But when I double click and install it, and open comfyui, the Manager button doesn't appear

**Comment by ltdrdata** [comment]
Show me the error messages on terminal.

**Comment by KevenLou321** [comment]
> 

![image](https://github.com/ltdrdata/ComfyUI-Manager/assets/141619997/69f30a0f-ff00-439a-b9db-d67808d91460)

**Comment by ltdrdata** [comment]
> > 
> 
> ![image](https://github.com/ltdrdata/ComfyUI-Manager/assets/141619997/69f30a0f-ff00-439a-b9db-d67808d91460)
> 

There is no error message. Is it all?

**Comment by KevenLou321** [comment]
> > > 
> > 
> > 
> > ![image](https://user-images.githubusercontent.com/141619997/258771722-69f30a0f-ff00-439a-b9db-d67808d91460.png)
> 
> There is no error message. Is it all?

yes I just install it very easily, but no manager button there

**Comment by ltdrdata** [comment]
> > > > 
> > > 
> > > 
> > > ![image](https://user-images.githubusercontent.com/141619997/258771722-69f30a0f-ff00-439a-b9db-d67808d91460.png)
> > 
> > There is no error message. Is it all?
> 
> yes I just install it very easily, but no manager button there
> 

Oh that is the installation log.
I meant startup log of ComfyUI. Sorry.

**Comment by KevenLou321** [comment]
> 

![image](https://github.com/ltdrdata/ComfyUI-Manager/assets/141619997/f5dd4634-a06b-4bc6-8a65-4e0af388f221)
XD

**Comment by KevenLou321** [comment]
D:\work\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable>.\python_embeded\python.exe -s ComfyUI\main.py --windows-standalone-build
Total VRAM 12288 MB, total RAM 32610 MB
xformers version: 0.0.20
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3060 : cudaMallocAsync
Using xformers cross attention
Traceback (most recent call last):
  File "D:\work\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\python_embeded\lib\site-packages\git\__init__.py", line 89, in <module>
    refresh()
  File "D:\work\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\python_embeded\lib\site-packages\git\__init__.py", line 76, in refresh
    if not Git.refresh(path=path):
  File "D:\work\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\python_embeded\lib\site-packages\git\cmd.py", line 392, in refresh
    raise ImportError(err)
ImportError: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\work\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1647, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\work\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\custom_nodes\update-fix.py", line 1, in <module>
    import git
  File "D:\work\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\python_embeded\lib\site-packages\git\__init__.py", line 91, in <module>
    raise ImportError("Failed to initialize: {0}".format(_exc)) from _exc
ImportError: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet


Cannot import D:\work\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\custom_nodes\update-fix.py module for custom nodes: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet


Import times for custom nodes:
   0.0 seconds: D:\work\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\custom_nodes\searge_sdxl_sampler_node.py
   0.0 seconds (IMPORT FAILED): D:\work\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\custom_nodes\update-fix.py

Starting server

To see the GUI go to: http://127.0.0.1:8188

**Comment by ltdrdata** [comment]
Yeah.. It's wrong...

Delete `D:\work\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\custom_nodes\update-fix.py`

And download this into `D:\work\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable`
https://github.com/ltdrdata/ComfyUI-Manager/raw/main/scripts/install-manager-for-portable-version.bat

And execute it.

Otherwise,

goto  
`D:\work\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\custom_nodes\`
in cmd 

execute 
`git clone https://github.com/ltdrdata/ComfyUI-Manager`

**Comment by dcary94** [comment]
Hi there,

I've followed the installation instructions but the manager button doesn't appear when I relaunch ComfyUI.

Any help would be greatly appreciated!

**Comment by KevenLou321** [comment]
> Hi there,
> 
> I've followed the installation instructions but the manager button doesn't appear when I relaunch ComfyUI.
> 
> Any help would be greatly appreciated!

i just solve the problem, 
should install this https://git-scm.com/downloads
and try install it again in cmd

**Comment by dcary94** [comment]
Right, so the reason I can't clone the repo to the ComfyUI repository via cmd and thereafter install the custom node is because I don't have 'Git' installed on my PC? - apologies, never heard of 'Git' and as you can tell, new to GitHub - Thanks!

**Comment by cn112cn** [comment]
Hmm, strange... I have git version 2.34.1 and still have this issue.

My installation seems to be completed properly:
root@C.6739850:/workspace/ComfyUI/custom_nodes$ git clone https://github.com/ltdrdata/ComfyUI-Manager.git
Cloning into 'ComfyUI-Manager'...
remote: Enumerating objects: 833, done.
remote: Counting objects: 100% (176/176), done.
remote: Compressing objects: 100% (105/105), done.
remote: Total 833 (delta 75), reused 93 (delta 66), pack-reused 657
Receiving objects: 100% (833/833), 2.07 MiB | 6.10 MiB/s, done.
Resolving deltas: 100% (524/524), done.

Any idea on what can be causing the issue?

I'm running ComfyUI 3.0 on Runpod.io Cloud GPU, this is the Docker image being used: https://hub.docker.com/r/runpod/stable-diffusion

**Comment by dcary94** [comment]
After installing the latest version 'Git' on my machine I was advised to install the 04.06.2023 version of comfyui as apparently there seem to be issues with other versions....not sure. Nevertheless this did work for me!![image](https://github.com/ltdrdata/ComfyUI-Manager/assets/85063788/6c448206-6dfd-4e85-9b92-d142d58ab0f0)

**Comment by ltdrdata** [comment]
> Hmm, strange... I have git version 2.34.1 and still have this issue.
> 
> My installation seems to be completed properly: root@C.6739850:/workspace/ComfyUI/custom_nodes$ git clone https://github.com/ltdrdata/ComfyUI-Manager.git Cloning into 'ComfyUI-Manager'... remote: Enumerating objects: 833, done. remote: Counting objects: 100% (176/176), done. remote: Compressing objects: 100% (105/105), done. remote: Total 833 (delta 75), reused 93 (delta 66), pack-reused 657 Receiving objects: 100% (833/833), 2.07 MiB | 6.10 MiB/s, done. Resolving deltas: 100% (524/524), done.
> 
> Any idea on what can be causing the issue?
> 
> I'm running ComfyUI 3.0 on Runpod.io Cloud GPU, this is the Docker image being used: https://hub.docker.com/r/runpod/stable-diffusion

I have identified an issue where the .js file is not being copied to the appropriate location in the runpod. I have added logging related to the file copying location in the latest ComfyUI-Manager. Please install the latest ComfyUI-Manager and provide the complete logs for the ComfyUI startup.

**Comment by cn112cn** [comment]
Thank you for taking the time to look into this.

I created a new instance, repeated the process and checked the logs.

The issue was that Torchvision was not installed on that Runpod image.

After running "pip install torchvision" the Manager is now visible.

**Comment by lizhengxuejohn** [comment]
Hi ,

I've followed the installation instructions to install ComfyUI Manager on my Ubuntu 22.04.1. There is no any issues appeared.  But the manager button doesn't appear when I relaunch ComfyUI.

Any help would be greatly appreciated!

**Comment by jayjake0** [comment]
> Thank you for taking the time to look into this.
> 
> I created a new instance, repeated the process and checked the logs.
> 
> The issue was that Torchvision was not installed on that Runpod image.
> 
> After running "pip install torchvision" the Manager is now visible.

Thanks for this workaround.

**Comment by sapientsnap** [comment]
Hello!
8 month Comfy user – I recently had to wipe my system of all Python instances (to solve conflicting path issues), then reinstalled Comfy and all dependencies properly in a venv. The only instance of Python and Comfy dependencies on my system currently is Python 3.11.9 residing inside the venv.

All is working except the manager. I’m having the issue where when I launch Comfy the manager and share buttons are not visible.  Help from my ComfyCommunity friends is greatly appreciated!!  

To address this message I’ve reinstalled Manager twice using two different methods to no avail.  Here is the full startup log followed by an error in the ComfyUI log

(venvSD) MoAI@Morales-M1-16 ComfyUI % python main.py
[START] Security scan
[DONE] Security scan
## ComfyUI-Manager: installing dependencies done.
** ComfyUI startup time: 2024-07-14 20:26:11.804393
** Platform: Darwin
** Python version: 3.12.4 (v3.12.4:8e8a4baf65, Jun  6 2024, 17:33:18) [Clang 13.0.0 (clang-1300.0.29.30)]
** Python executable: /Users/MoAI/AI/StableDiffusion/venvSD/bin/python3
** ComfyUI Path: /Users/MoAI/AI/StableDiffusion/ComfyUI
** Log path: /Users/MoAI/AI/StableDiffusion/ComfyUI/comfyui.log

Prestartup times for custom nodes:
   0.3 seconds: /Users/MoAI/AI/StableDiffusion/ComfyUI/custom_nodes/ComfyUI-Manager

Total VRAM 16384 MB, total RAM 16384 MB
pytorch version: 2.3.1
Set vram state to: SHARED
Device: mps
Using sub quadratic optimization for cross attention, if you have memory or speed issues try using: --use-split-cross-attention
### Loading: ComfyUI-Manager (V2.46.3)
### ComfyUI Revision: 2362 [a3dffc44] | Released on '2024-07-13'

Import times for custom nodes:
   0.0 seconds: /Users/MoAI/AI/StableDiffusion/ComfyUI/custom_nodes/websocket_image_save.py
   0.1 seconds: /Users/MoAI/AI/StableDiffusion/ComfyUI/custom_nodes/ComfyUI-Manager

Starting server

To see the GUI go to: http://127.0.0.1:8188
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/github-stats.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/model-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/alter-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json



**Here is an error listed in the console log from Comfy:**

console
error
20:40:18.516Z
[
  "Error loading extension",
  "/extensions/ComfyUI-Manager/components-manager.js",
  "SyntaxError\nImporting module '../../scripts/ui/draggableList.js' is not found.\n"
]

**Comment by benr** [comment]
I just ran into this problem after an installation on MacOS.  The solution was to manually install it: 

```
## Checkout the ComfyUI-Manager
cd custom_nodes/
git clone https://github.com/ltdrdata/ComfyUI-Manager

## Then start ComfyUI
cd ..
 python3 main.py
```

**Comment by sapientsnap** [comment]
> I just ran into this problem after an installation on MacOS. The solution was to manually install it:
> 
> ```
> ## Checkout the ComfyUI-Manager
> cd custom_nodes/
> git clone https://github.com/ltdrdata/ComfyUI-Manager
> 
> ## Then start ComfyUI
> cd ..
>  python3 main.py
> ```

Thank you @benr . I have clean-installed Comfy and Manager several times using several different methods. What I learned is that mine is a browser cache problem. Out of frustration the other day, I refreshed my browser like a machine gun, hitting the refresh about a dozen times in rapid succession and guess what happened?… The manager appeared. 

😳

When IU refreshed again, it disappeared. 😡

Now, I’ve learned that when I launch, the manager never shows up, but if I refresh 5~50 times, it will appear.  Very weird.  This is on both DuckDuckGo and Safari. Today I just simply cleared the browser cache and it launched properly the first time.

I wish I knew a few days ago it was a browser problem before spending countless hours troubleshooting in a different direction. To my credit, I did consider the browser issue early on, but only refreshing once or twice masked the issue which is resolved only by clearing the cache or reloading the page a billion times. 🙄

Thanks again for your suggestion. It’s great to get a reply when shooting up a flare.
